Tests/HAPTests/EndpointTests.swift

```diff
@@ -143,7 +145,7 @@ class EndpointTests: XCTestCase {
             XCTAssertEqual(nameCharacteristic["value"] as? String, "Night stand left")
         }
 
-        do {
+        testQueue.async {
```

Comment:
1. How do we ensure correct ordering of multiple items posted on the queue?
2. How do we ensure we actually fail the tests from something happening on another queue?
3. Howe do we ensure the async tests actually finish before finishing the test suite?

Reply:
1. The queue is serial, so although each test is run asynchronously, they are executed in sequence. 
2. The call to `application(...)` is run synchronously on the main queue, (from `characteristic(...)`) and returns an `HTTPResponse` back to the test queue where it is examined for correctness. i.e the tests are guaranteed to be performed after the `application(...)` call is complete.
3. We can add the following before the end of each test function
```
        let expectation = self.expectation(description: "Test Complete")
        testQueue.async { expectation.fulfill() }
        waitForExpectations(timeout: 5, handler: nil)
```

BTW I've not dug in to NIO architecture yet, but a better design would be to have NIO call `application()` asynchronously and return a promise to process the result once complete.

Reply:
Ok; looks good to me!